packages/core/bootstrap/src/lib/external-adapter/errors.ts

```diff
@@ -30,7 +30,7 @@ export class AdapterError extends Error {
     this.name = name
     this.message = message
     this.cause = cause
-    this.endpoint = endpoint!
+    this.url = url!
```

Comment:
I don't think we should use non-null assertions like this.
Since `url` isn't passed in on every `AdapterError` it is actually functioning as optional.
We should either make it so in the static property types, or ensure it is actually non-null everywhere (which may be a lot of work).